File: programs/class3/matrix.f90
program matrix_multiply
  implicit none
  integer,parameter :: n=1000
  real*8 mat1(n,n),mat2(n,n),prod(n,n)
  integer k,i,j
  real*8 start_time,stop_time

  ! Initialing mat1, mat2
  do i=1,n
     do j=1,n
       mat1(i,j)=i/j
       mat2(i,j)=j/i
    enddo
  enddo

  !Multiply matrix
  call cpu_time(start_time)
  prod=0.0
  do i=1,n
    do j=1,n
      do k=1,n
        prod(i,j)=prod(i,j)+mat1(i,k)*mat2(k,j)
      enddo
    enddo
  enddo
  call cpu_time(stop_time)
  write(*,*)'time required in loop',stop_time-start_time

  call cpu_time(start_time)
  prod=matmul(mat1,mat2)
  call cpu_time(stop_time)
  write(*,*) 'time required in matmul',stop_time-start_time

end program matrix_multiply

File: programs/class4/matrix.py
#This is a program to write matrix multiplication

#Take a 3*3
A=[[12,7,3],
  [4,5,6],
  [7,8,9]]

#Take a 3*4 Matrix
B=[[5,8,1,2],
   [6,7,3,0],
   [4,5,9,1]]
C=[[0,0,0,0],
   [0,0,0,0],
   [0,0,0,0]]
print (len(B[0]))
for i in range(len(A)):
   for j in range (len(B[1])):
      for k in range(len(B)):
         C[i][j]+=A[i][k]*B[k][j]
print ('product matrix ')
print (C)
